app/src/main/java/net/gsantner/markor/format/csv/CsvTextConverter.java

```diff
@@ -0,0 +1,186 @@
+import java.io.Closeable;
+import java.io.File;
+import java.io.IOException;
+import java.io.Reader;
+import java.io.StringReader;
+
+import other.de.stanetz.jpencconverter.JavaPasswordbasedCryption;
+
+/**
+ * Part of Markor-Architecture implementing Preview/Export for csv.
+ * <p>
+ * Converts csv to md and let
+ * {@link  MarkdownTextConverter#convertMarkup(String, Context, boolean, File)}
+ * do the rest.
+ * <p>
+ * This way csv columns may contain md expressions like bold text.
+ */
+@SuppressWarnings("WeakerAccess")
+public class CsvTextConverter extends MarkdownTextConverter {
+    @Override
+    public String convertMarkup(String csvMarkup, Context context, boolean isExportInLightMode, File file) {
+        String mdMarkup = Csv2MdTable.toMdTable(csvMarkup);
+        return super.convertMarkup(mdMarkup, context, isExportInLightMode, file);
+    }
+
+    @Override
+    protected boolean isFileOutOfThisFormat(String filepath, String extWithDot) {
+        filepath = filepath.replace(JavaPasswordbasedCryption.DEFAULT_ENCRYPTION_EXTENSION, "");
+        return filepath.toLowerCase().endsWith(".csv");
+    }
+
+    protected static class Csv2MdTable implements Closeable {
+        public static final int BUFFER_SIZE = 8096;
+
+        private static final String MD_LINE_DELIMITER = "\n";
+        private static final String MD_COL_DELIMITER = "|";
+        private static final String MD_HEADER_LINE_DELIMITER = MD_COL_DELIMITER + ":---";
+        private final CSVReader m_csvReader;
+        private int m_lineNumber = 0;
+
+        private Csv2MdTable(CsvConfig csvConfig, Reader csvDataReader) {
+            ICSVParser parser = new CSVParserBuilder()
+                    .withSeparator(csvConfig.getFieldDelimiterChar())
+                    .withQuoteChar(csvConfig.getQuoteChar())
+                    .build();
+            m_csvReader = new CSVReaderBuilder(csvDataReader)
+                    .withSkipLines(0)
+                    .withCSVParser(parser)
+                    .withKeepCarriageReturn(true)
+                    .build();
+        }
+
+        public static String toMdTable(String csvMarkup) {
+            // parser cannot handle empty lines if they are not "\r\n"
+            return toMdTable(new StringReader(csvMarkup.replace("\n", "\r\n")));
+        }
+
+        public static String toMdTable(Reader csvMarkup) {
+            StringBuilder mdMarkup = new StringBuilder();
+            try (BufferedReader bufferedReader = new BufferedReader(csvMarkup, BUFFER_SIZE)) {
+                CsvConfig csvConfig = inferCsvConfiguration(bufferedReader);
+                try (Csv2MdTable toMdTable = new Csv2MdTable(csvConfig, bufferedReader)) {
+                    String[] headers = toMdTable.readNextCsvColumnLine();
+
+                    if (headers != null && headers.length > 0) {
+                        addColumnsLine(mdMarkup, headers, headers.length);
+
+                        for (int i = 0; i < headers.length; i++) {
```

Comment:
foreach can be used here

for (final String h : headers) {

Reply:
done